src/main/java/seedu/address/model/internship/Status.java

```diff
@@ -0,0 +1,85 @@
+package seedu.address.model.internship;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+/**
+ * Represents a Status in the InternshipItem. Each status is tagged to a date of creation.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Status {
+
+    // Identity field
+    private final JobStatus jobStatus;
+
+    // Data fields
+    private final Date date;
```

Comment:
I suppose having this field means in the UI, we'll have some text like "As of: `date`"?

Reply:
Kind of, the date will be used on the main display as well as on the right pane:
![image](https://user-images.githubusercontent.com/65328119/94686049-d3c96d80-035c-11eb-8541-23901a7c163e.png)
![image](https://user-images.githubusercontent.com/65328119/94686154-02474880-035d-11eb-813e-77df8fbbb672.png)

Reply:
Ah okay! My bad for missing this out, thanks for highlighting!